examples/with-eip-1193-provider/src/lib/turnkey.ts

```diff
@@ -0,0 +1,176 @@
+"use server";
+
+import {
+  ApiKeyStamper,
+  type TApiKeyStamperConfig,
+} from "@turnkey/api-key-stamper";
+import { TurnkeyClient, createActivityPoller } from "@turnkey/http";
+import { Attestation, Email, PassKeyRegistrationResult } from "./types";
+
+import { generateRandomBuffer, base64UrlEncode } from "./utils";
+import {
+  ETHEREUM_WALLET_DEFAULT_PATH,
+  PUBKEY_CRED_TYPE,
+  ALG_ES256,
+} from "./constants";
+import { UUID } from "crypto";
+import { Address } from "viem";
+
+const {
+  TURNKEY_API_PUBLIC_KEY,
+  TURNKEY_API_PRIVATE_KEY,
+  NEXT_PUBLIC_ORGANIZATION_ID,
+  NEXT_PUBLIC_TURNKEY_RPID,
+} = process.env;
+
+export const registerPassKey = async (
+  email: Email
+): Promise<PassKeyRegistrationResult> => {
+  if (!NEXT_PUBLIC_TURNKEY_RPID) {
+    throw "Error must define NEXT_PUBLIC_TURNKEY_RPID in your .env file";
+  }
+
+  // @todo - Add error handling
+  const { getWebAuthnAttestation } = await import("@turnkey/http");
+  const challenge = generateRandomBuffer();
+  const authenticatorUserId = generateRandomBuffer();
+  const user = email.split("@")[0];
+  // An example of possible options can be found here:
+  // https://www.w3.org/TR/webauthn-2/#sctn-sample-registration
```

Comment:
Unfortunately I think this is erroring out? I tried to run this demo and got:
```
 ⨯ ReferenceError: window is not defined
    at hasWebAuthnSupport (../../packages/http/dist/webauthn.mjs:100:5)
    at getWebAuthnAttestation (../../packages/http/dist/webauthn.mjs:89:31)
    at $$ACTION_0 (./src/lib/turnkey.ts:39:31)
    at async $$ACTION_3 (./src/lib/turnkey.ts:145:39)
  87 | // https://web.dev/articles/passkey-form-autofill#feature-detection, https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredential
  88 | function hasWebAuthnSupport() {
> 89 |     return !!window.PublicKeyCredential;
     |    ^
  90 | }
  91 |
  92 | export { getWebAuthnAssertion, getWebAuthnAttestation, protocolTransportEnumToInternalEnum };
 ⨯ ../packages/http/dist/webauthn.mjs (89:4) @ hasWebAuthnSupport
 ⨯ ReferenceError: window is not defined
    at hasWebAuthnSupport (../../packages/http/dist/webauthn.mjs:100:5)
    at getWebAuthnAttestation (../../packages/http/dist/webauthn.mjs:89:31)
    at $$ACTION_0 (./src/lib/turnkey.ts:39:31)
    at async $$ACTION_3 (./src/lib/turnkey.ts:145:39)
  87 | // https://web.dev/articles/passkey-form-autofill#feature-detection, https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredential
  88 | function hasWebAuthnSupport() {
> 89 |     return !!window.PublicKeyCredential;
     |    ^
  90 | }
```

Maybe because this is rendered server side somehow?